lib/fast_cache/cache.rb

```diff
@@ -194,7 +198,9 @@ def store(key, val)
     end
 
     def store_entry(key, entry)
-      @data.delete(key)
+      found = true
+      old_entry = @data.delete(key) { found = false }
+      @cleanup.call(old_entry.value) if @cleanup && found
```

Comment:
It's been a while since I wrote this, but if I recall correctly, the reason for this was to properly handle the case where a `nil` or `false` value was found in (and deleted from) the cache.